mixins/localize/localize-mixin.js

```diff
@@ -0,0 +1,288 @@
+			formattedMessage = translatedMessage.format(params);
+		} catch (e) {
+			console.error(e);
+		}
+
+		return formattedMessage;
+	}
+
+	localizeHTML(key, params = {}) {
+
+		const { language, value } = this.__resources?.[key] ?? {};
+		if (!value) return '';
+
+		const translatedMessage = new IntlMessageFormat(value, language);
+		let formattedMessage = value;
+		try {
+			formattedMessage = translatedMessage.format({
+				b: chunks => localizeMarkup`<strong>${chunks}</strong>`,
+				br: () => localizeMarkup`<br>`,
+				em: chunks => localizeMarkup`<em>${chunks}</em>`,
+				i: chunks => localizeMarkup`<em>${chunks}</em>`,
+				p: chunks => localizeMarkup`<p>${chunks}</p>`,
+				strong: chunks => localizeMarkup`<strong>${chunks}</strong>`,
+				...params
+			});
+			validateMarkup(formattedMessage);
+		} catch (e) {
+			console.error(e);
+		}
+
+		return formattedMessage;
+	}
+
+	static _generatePossibleLanguages(config) {
+
+		if (config?.useBrowserLangs) return navigator.languages.map(e => e.toLowerCase()).concat('en');
+
+		const { language, fallbackLanguage } = this.documentLocaleSettings;
+		const langs = [ language, fallbackLanguage ]
+			.filter(e => e)
+			.map(e => [ e.toLowerCase(), e.split('-')[0] ])
+			.flat();
+
+		return Array.from(new Set([ ...langs, 'en-us', 'en' ]));
+	}
+
+	static _getAllLocalizeResources(config = this.localizeConfig) {
+		let resourcesLoadedPromises = [];
+		const superCtor = Object.getPrototypeOf(this);
+		// get imported terms for each config, head up the chain to get them all
+		if ('_getAllLocalizeResources' in superCtor) {
+			const superConfig = Object.prototype.hasOwnProperty.call(superCtor, 'localizeConfig') && superCtor.localizeConfig.importFunc ? superCtor.localizeConfig : config;
+			resourcesLoadedPromises = superCtor._getAllLocalizeResources(superConfig);
+		}
+		if (Object.prototype.hasOwnProperty.call(this, 'getLocalizeResources') || Object.prototype.hasOwnProperty.call(this, 'resources')) {
+			const possibleLanguages = this._generatePossibleLanguages(config);
+			const res = this.getLocalizeResources(possibleLanguages, config);
+			resourcesLoadedPromises.push(res);
+		}
+		return resourcesLoadedPromises;
+	}
+
+	_hasResources() {
+		return this.constructor['getLocalizeResources'] !== undefined;
+	}
+
+	_onResourcesChange() {
+		/** @ignore */
+		this.dispatchEvent(new CustomEvent('d2l-localize-resources-change'));
+	}
+
+});
+
+export const LocalizeDynamicMixin = superclass => class extends LocalizeMixin(superclass) {
+
+	static async getLocalizeResources(langs, { importFunc, osloCollection, useBrowserLangs }) {
+
+		// in dev, don't request unsupported langpacks
+		if (!importFunc.toString().includes('switch') && !useBrowserLangs) {
+			langs = langs.filter(lang => supportedLangpacks.includes(lang));
+		}
+
+		for (const lang of [...langs, fallbackLang]) {
+
+			const resources = await importFunc(lang).catch(() => {});
```

Comment:
I think this is what we want
```suggestion
			const resources = await Promise.resolve(importFunc(lang)).catch(() => {})
```